chapter09/index.adoc

```diff
@@ -0,0 +1,451 @@
+. Open the Jaeger UI at http://localhost:16686[http://localhost:16686].
+. Search for traces associated with your application and confirm that the telemetry data is visible.
+
+=== Switching to Tracing Providers
+
+MicroProfile Telemetry supports multiple tracing providers, for exporting and analyzing traces. The default tracing provider is Jaeger, but developers can also use other providers such as Zipkin or OpenCensus.
+
+To switch to another tracing provider, replace the Jaeger dependency with the appropriate exporter dependency, such as *Zipkin*, and update the configuration properties accordingly.
+
+=== Context Propagation and Correlation
+
+*Context propagation* refers to the mechanism of carrying trace-related metadata, such as *trace IDs* and *span IDs*, across service and thread boundaries. This ensures that all spans created during a request can be linked together to form a complete trace.
+
+*How it works* 
+
+. *Trace Context*: Metadata that includes the +traceId+, +spanId+, and sampling information.
+. *Propagation Mechanisms*: Trace context is typically carried in HTTP headers (e.g., +traceparent+) or message properties in message queues.
+. *MicroProfile Integration*: MicroProfile Context Propagation seamlessly integrates with OpenTelemetry to ensure that the trace context is maintained across service calls.
+
+*Example: Propagating Context Across HTTP Requests*
+
+When making an HTTP request, the trace context is propagated using headers:
+
+[source, java]
+----
+import jakarta.ws.rs.client.Client;
+import jakarta.ws.rs.client.ClientBuilder;
+
+Client client = ClientBuilder.newClient();
+client.target("http://inventory-service/api/check")
+      .request()
+      .header("traceparent", "00-4bf92f3577b34da6a3ce929d0e0e4736-00f067aa0ba902b7-01")
+      .get();
+----
+
+In this example, the +traceparent+ header ensures that the trace context is passed to the downstream +inventory-service+.
+
+==== Correlation
+
+*Correlation* is the process of associating related spans and traces across multiple services and threads to form a cohesive view of a transaction. Correlation enables developers to:
+
+* Identify the source of bottlenecks or errors in distributed systems.
+* Understand the dependencies and interactions between services.
+
+==== Trace and Span IDs
+
+* *Trace ID*: A unique identifier shared across all spans in a single trace.
+* *Span ID*: A unique identifier for a single span. It is linked to a parent span, forming a hierarchy.
+
+*Example: Correlating Logs with Traces*
+
+By including trace and span IDs in logs, you can correlate logs with traces to gain deeper insights:
+
+[source, java]
+----
+import org.slf4j.MDC;
+
+MDC.put("traceId", "4bf92f3577b34da6a3ce929d0e0e4736");
+MDC.put("spanId", "00f067aa0ba902b7");
+
+log.info("Fetching product details for productId=12345");
+----
+
+When viewing logs, the +traceId+ and +spanId+ allow you to link specific log entries to the corresponding spans in your tracing system.
```

Comment:
I'm not familiar with MDC in slf4j, but OpenTelemetry includes support for logging and will capture the current context information automatically when something is logged.

Not being familiar with MDC here, I'm not sure if this example is good to have here, or whether it would be more helpful to talk about OpenTelemetry's native support for logging.

If we did keep the current example, it would probably be helpful to include how to extract the traceId and spanId from the current context.